### PR TITLE
fix: sp-split-button more button label

### LINKED ISSUE #3305: [Bug]: sp-split-button more button should be labelled by button element
### Code of conduct

- [X] I agree to follow this project's code of conduct.

### Impacted component(s)

sp-split-button

### Expected behavior

The popup button part of the split button should be labelled by the button part

### Actual behavior

It is currently labelled a hardcoded "More"

### Screenshots

_No response_

### What browsers are you seeing the problem in?

Firefox, Chrome, Safari, Microsoft Edge

### How can we reproduce this issue?

1. Go to https://opensource.adobe.com/spectrum-web-components/components/split-button/#sizes
2. Right Click on the More button and inspect the Accessibility Properties

### Sample code that illustrates the problem

_No response_

### Logs taken while reproducing problem

_No response_

### PR DESCRIPTION
**Pull request by jnurthen**
Fixes #3305

## Description

Changes the label for the more button from static text to a reference to the sibling button element

## Related issue(s)

#3305
-

## How has this been tested?

-   [x] _Test case 1_
    1. Go to https://jnurthen-issue3305--spectrum-web-components.netlify.app/components/split-button/
    2. Inspect the more part of the split button and look at the Accessible Name in Dev tools
    3. Note that it should be the same as the accessible name of the button part of the split button

## Types of changes

-   [x] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)
-   [ ] Chore (minor updates related to the tooling or maintenance of the repository, does not impact compiled assets)

## Checklist

-   [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
-   [x] My code follows the code style of this project.
-   [ ] If my change required a change to the documentation, I have updated the documentation in this pull request.
-   [x] I have read the **[CONTRIBUTING](<(https://github.com/adobe/spectrum-web-components/blob/main/CONTRIBUTING.md)>)** document.
-   [x] I have added tests to cover my changes.
-   [x] All new and existing tests passed.
-   [x] I have reviewed at the Accessibility Practices for this feature, see: [Aria Practices](https://www.w3.org/TR/wai-aria-practices/)

### CODE REVIEW

**Comment by Westbrook** [comment]
Firefox appears to ship a slightly different accessibility tree in response to these updates. Can you take a look at the difference and update either the implementation or the test case here?

File: packages/split-button/test/index.ts [hunk 1]
```diff
@@ -97,6 +98,73 @@ export function runSplitButtonTests(
...
+        const snapshot = (await a11ySnapshot(
+            {}
+        )) as unknown as NamedRoledPopupNode & {
+            children: NamedRoledPopupNode[];
+        };
+        expect(
+            findAccessibilityNode<NamedRoledPopupNode>(
+                snapshot,
+                (node) =>
+                    (node.role === 'button' || node.role === 'buttonmenu') &&
```
Comment (Westbrook) [inline packages/split-button/test/index.ts:130]: @jnurthen this gets the test passing in Firefox. Not sure if this counts as a _bug_ in Firefox (or possibly in Playwright, they're deprecating these API anyways 😢), but it seems to be OK on our end.

If it's good for you, I'm good to merge this in.

Comment (jnurthen) [reply packages/split-button/test/index.ts:130]: thanks for fixing that. That looks good.

**Review by Westbrook — APPROVED** [review]
LGTM. Thanks for getting this cleared up for us!

**Pull request merged**